# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546469984 28800
#      Wed Jan 02 14:59:44 2019 -0800
# Branch stable
# Node ID 2408f3b6a6db7b637b8db30da3dcc4299b714985
# Parent  3c5a676cd95b58bc227336de034e3c35fd4f371c
ov-intx.h: Fix static analyzer detected issues (bug #55347).

* ov-intx.h (double_value, float_value): Declare retval, but don't do
unnecessary initialization.

diff --git a/libinterp/octave-value/ov-intx.h b/libinterp/octave-value/ov-intx.h
--- a/libinterp/octave-value/ov-intx.h
+++ b/libinterp/octave-value/ov-intx.h
@@ -102,33 +102,33 @@ public:
   uint32_array_value (void) const { return uint32NDArray (matrix); }
 
   uint64NDArray
   uint64_array_value (void) const { return uint64NDArray (matrix); }
 
   double
   double_value (bool = false) const
   {
-    double retval = lo_ieee_nan_value ();
+    double retval;
 
     if (isempty ())
       err_invalid_conversion (type_name (), "real scalar");
 
     warn_implicit_conversion ("Octave:array-to-scalar",
                               type_name (), "real scalar");
 
     retval = matrix(0).double_value ();
 
     return retval;
   }
 
   float
   float_value (bool = false) const
   {
-    float retval = lo_ieee_float_nan_value ();
+    float retval;
 
     if (isempty ())
       err_invalid_conversion (type_name (), "real scalar");
 
     warn_implicit_conversion ("Octave:array-to-scalar",
                               type_name (), "real scalar");
 
     retval = matrix(0).float_value ();
